chainer/training/extensions/parameter_statistics.py

```diff
@@ -0,0 +1,233 @@
+
+        if not isinstance(links, (tuple, list)):
+            links = links,
+
+        self._links = links
+        self._trigger = training.trigger.get_trigger(trigger)
+        self._prefix = prefix
+        self._summary = reporter.DictSummary()
+        self._targets = [('W', 'data'), ('b', 'data'),
+                         ('W', 'grad'), ('b', 'grad')]
+        self._sparsity_targets = []
+
+        if sparsity:
+            if sparsity_include_bias:
+                self._sparsity_targets.append((('W', 'b'), 'data'))
+            else:
+                self._sparsity_targets.append((('W'), 'data'))
+
+        self._statistic_functions = ('min', 'max', 'mean', 'std')
+        self._percentile_sigmas = (0.13, 2.28, 15.87, 50, 84.13, 97.72, 99.87)
+
+    def __call__(self, trainer):
+        """Execute the statistics extension.
+
+        Collect statistics for the current state of parameters.
+
+        Note that this method will merely update its statistic summary, unless
+        the internal trigger is fired. If the trigger is fired, the summary
+        will also be reported and then reset for the next accumulation.
+
+        Args:
+            trainer (~chainer.training.Trainer): Associated trainer that
+                invoked this extension.
+        """
+        for link in self._links:
+            for target in self._targets:
+                stats = self.get_statistics(link, *target)
+                stats = self.post_process(stats)
+                self._summary.add(stats)
+
+            for target in self._sparsity_targets:
+                stats = self.get_sparsity(link, *target)
+                stats = self.post_process(stats)
+                self._summary.add(stats)
+
+        if self._trigger(trainer):
+            reporter.report(self._summary.compute_mean())
+            self._summary = reporter.DictSummary()  # Clear summary
+
+    def post_process(self, stats):
+        """Post processing of the data before adding them to the summary."""
+        if self._prefix is not None:
+            _prefix_statistics(self._prefix, stats)
+
+        return stats
+
+    def statistics(self, params):
+        return _statistics(params, self._statistic_functions)
+
+    def percentiles(self, params):
+        return _percentiles(params, self._percentile_sigmas)
+
+    def sparsity(self, params):
+        return _sparsity(params)
+
+    def get_statistics(self, link, param_names, attr_names):
+
+        key = _statistic_key(link, param_names, attr_names)
+        params = _flatten_link(link, param_names, attr_names)
+        stats = {}
+
+        for f, s in self.statistics(params).items():
+            stats['{}/{}'.format(key, f)] = s
+
+        for i, p in enumerate(self.percentiles(params)):
+            stats['{}/percentile/{}'.format(key, i)] = p
+
+        return stats
+
+    def get_sparsity(self, link, param_names, attr_names):
+
+        key = _statistic_key(link, param_names, attr_names)
+        key += '/zeros'
```

Comment:
How about using `zeros` in other names as this key suggests? I feel zeros is clearer than sparsity.